### PR TITLE
Feat(generator_proxy_template): start type checking

### PR DESCRIPTION
**Pull request by ShadoySV**

### CODE REVIEW

File: contracts/tokenomics/generator_proxy_to_mirror/src/state.rs [hunk 1]
```diff
@@ -8,15 +8,15 @@ use cw_storage_plus::Item;
 /// This structure holds the main parameters for the of generator_proxy_to_mirror contract.
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 pub struct Config {
-    /// The generator contract address
+    /// The Generator contract address
     pub generator_contract_addr: Addr,
-    /// The Astroport pair contract address
+    /// The target Astroport pair contract address
     pub pair_addr: Addr,
     /// The contract address for the Astroport MIR LP token
```
Comment (epanchee) [inline contracts/tokenomics/generator_proxy_to_mirror/src/state.rs:15]: Remove MIR

File: templates/generator_proxy_template/src/contract.rs [hunk 2]
```diff
@@ -94,39 +94,35 @@ pub fn execute(
...
+    let response = Response::new();
     let cfg = CONFIG.load(deps.storage)?;
 
     if let Ok(Cw20HookMsg::Deposit {}) = from_binary(&cw20_msg.msg) {
         if cw20_msg.sender != cfg.generator_contract_addr || info.sender != cfg.lp_token_addr {
             return Err(ContractError::Unauthorized {});
         }
-        // deposit tokens in the end reward contract
-        unimplemented!()
+        // TODO: deposit tokens in the end reward contract here
```
Comment (epanchee) [inline templates/generator_proxy_template/src/contract.rs:117]: Better to use macros `todo!("deposit tokens in the end reward contract here")` so the developer will know that something needs to be implemented

File: templates/generator_proxy_template/src/contract.rs [hunk 3]
```diff
@@ -140,34 +136,36 @@ fn update_rewards(deps: DepsMut) -> Result<Response, ContractError> {
         .push(SubMsg::new(CosmosMsg::Wasm(WasmMsg::Execute {
             contract_addr: cfg.reward_contract_addr.to_string(),
             funds: vec![],
-            // specify a withdraw rewards message that withdraws rewards from the end reward contract
-            msg: unimplemented!(),
+            // TODO: specify a withdraw rewards message that withdraws rewards from the end reward contract here
+            msg: to_binary(&())?,
```
Comment (epanchee) [inline templates/generator_proxy_template/src/contract.rs:140]: Here macros is also preferable

File: templates/generator_proxy_template/src/contract.rs [hunk 4]
```diff
@@ -226,8 +227,7 @@ fn withdraw(
         res.balance
     };
 
-    // withdraw from the 3rd party reward contract
-    unimplemented!();
+    // TODO: withdraw from the 3rd party reward contract here
```
Comment (epanchee) [inline templates/generator_proxy_template/src/contract.rs:230]: todo("...") here as well

**Review by maksk7 — APPROVED** [review]

File: contracts/tokenomics/generator_proxy_to_mirror/src/state.rs (hunk 1 above)
Comment (ShadoySV) [reply contracts/tokenomics/generator_proxy_to_mirror/src/state.rs:15]: Why? It's the proxy to mirror, not the proxy template

File: templates/generator_proxy_template/src/contract.rs (hunk 2 above)
Comment (ShadoySV) [reply templates/generator_proxy_template/src/contract.rs]: Done. Thanks

File: contracts/tokenomics/generator_proxy_to_mirror/src/state.rs (hunk 1 above)
Comment (epanchee) [reply contracts/tokenomics/generator_proxy_to_mirror/src/state.rs:15]: Ah, yeah. Sorry

**Review by epanchee — APPROVED** [review]

**Pull request merged**